src/data_projection_routines.f90

```diff
@@ -1188,6 +1197,7 @@ SUBROUTINE DataProjection_DataPointsProjectionEvaluate(DATA_PROJECTION,PROJECTIO
                 ALLOCATE(PROJECTED_DISTANCE(2,NUMBER_OF_DATA_POINTS),STAT=ERR) !PROJECTED_DISTANCE(2,:) stores the compuational node number, the information for each data point has to be stored in the corresponding column for MPI_ALLREDUCE with location return to work
                 IF(ERR/=0) CALL FLAG_ERROR("Could not allocate projected distance.",ERR,ERROR,*999)
                 ALLOCATE(PROJECTED_XI(DATA_PROJECTION%NUMBER_OF_XI,NUMBER_OF_DATA_POINTS),STAT=ERR) !the information for each data point is stored in the corresponding column to be consistent with PROJECTED_DISTANCE
+                ALLOCATE(PROJECTION_VECTORS(DATA_PROJECTION%COORDINATE_SYSTEM_DIMENSIONS, NUMBER_OF_DATA_POINTS), STAT=ERR) !ZJW: The projection vector for each data point is stored in the corresponding colum to be consistent with PROJECTED_DISTANCE. 
                 IF(ERR/=0) CALL FLAG_ERROR("Could not allocate projected.",ERR,ERROR,*999)
```

Comment:
If projection vector is part of the projection result type then why are we allocating projection vectors here?